sdk/helper/keysutil/policy.go

```diff
@@ -1606,6 +1578,54 @@ func verifyWithECDSA(keyType KeyType, keyParams KeyEntry, input, sigBytes []byte
 	return ecdsa.Verify(key, input, ecdsaSig.R, ecdsaSig.S), nil
 }
 
+func (p *Policy) verifyEd25519WithOptions(ver int, input []byte, context []byte, publicKey string, options *SigningOptions, sigBytes []byte) (bool, error) {
+	var pub ed25519.PublicKey
+
+	if publicKey == "" {
```

Comment:
Hmm I don't think we can add this if statement here as it will break verification with key derivation enabled on the policy as you'll use the main public key and not the derived key?